WordPress/gutenberg
# Close block settings menu after removing block

## Description

Fixes #15458 by closing the block settings menu after removing a block through that menu.
Also fixes #15313.

This prevents editor crashes when there are no more blocks left but the block settings menu is still rendered as it's open.

## How has this been tested?

Following the steps outlined in #15458 I did this:

1. Enabled fixed toolbar
2. Create block
3. Click on "More options" to open block settings menu
4. Remove block
5. Verify that the menu is closed and no errors are thrown in the console and the editor is also still functioning.

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
